### PR TITLE
[MM-59825] Update module names for mono repo

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
When we updated the module name for the mono repo to `github.com/mattermost/mattermost/server/v8`, it broke a lot of linters. 210cf04e2c8930591a1ceefaee66a7ef44b7db32 fixes that.

2b854aceadc5f39c2bd12a1524d2653ba35e7608 adds tests for the `errorAssertions` linter, which didn't have any before. Please note that the tests do not protect against broken module names.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59825

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,5 +6,5 @@ require (
 	github.com/getkin/kin-openapi v0.61.0
 	github.com/pkg/errors v0.9.1
 	github.com/sajari/fuzzy v1.0.0
-	golang.org/x/tools v0.0.0-20191127201027-ecd32218bd7f
+	golang.org/x/tools v0.23.0
```
Comment (hanzei) [inline go.mod:9]: Without updating `golang.org/x/tools`, all tests break on my machine.

**Review by agnivade — APPROVED** [review]
Wow thank you for catching this Ben!

**Review by streamer45 — APPROVED** [review]
Nice work!

**Comment by hanzei** [comment]
I tried updating the tests to use an actual import of `github.com/mattermost/mattermost/server/public` using a `go.mod` file in the `testdata` directory. Locally the tests run fine, but on CI they failed because the dependencies in the separate `go.mod` file didn't get downloaded.

I will try to resolve that error in a separate PR.

**Pull request merged**